core/experiments/manager.php

```diff
@@ -100,13 +100,34 @@ public function add_feature( array $options ) {
 			$experimental_data['state'] = self::STATE_DEFAULT;
 		}
 
+		if ( ! empty( $experimental_data['dependency'] ) ) {
+			foreach ( $experimental_data['dependency'] as $key => $dependency ) {
+				if ( ! class_exists( $dependency ) ) {
+					throw new \Exception( sprintf( 'Dependency class %s not found', $dependency ) );
+				}
+
+				$experimental_data['dependency'][ $key ] = $dependency::instance();
+			}
+		}
+
 		$this->features[ $options['name'] ] = $experimental_data;
 
 		if ( $experimental_data['mutable'] && is_admin() ) {
 			$feature_option_key = $this->get_feature_option_key( $options['name'] );
 
-			$on_state_change_callback = function( $old_state, $new_state ) use ( $experimental_data ) {
-				$this->on_feature_state_change( $experimental_data, $new_state );
+			$on_state_change_callback = function( $old_state, $new_state ) use ( $experimental_data, $feature_option_key ) {
+				try {
+					$this->on_feature_state_change( $experimental_data, $new_state, $feature_option_key );
+				} catch ( Exceptions\Dependency_Exception $e ) {
+					$message = sprintf(
+						'<p>%s</p> <p><a href="#" onclick="%s">%s</a></p>',
+						esc_html__( $e->getMessage(), 'elementor' ), // phpcs:ignore
```

Comment:
Not sure you can translate an exception message. It should be translated in the exception itself.

Reply:
Handled.